framework/src/node/consensus/synchronizer/block_synchronization_mechanism.ts

```diff
@@ -96,11 +93,12 @@ export class BlockSynchronizationMechanism extends BaseSynchronizer {
 	public async isValidFor(): Promise<boolean> {
 		// 2. Step: Check whether current chain justifies triggering the block synchronization mechanism
 		const finalizedBlock = await this._chain.dataAccess.getBlockHeaderByHeight(
-			this.bft.finalizedHeight,
+			this.blockExecutor.getFinalizedHeight(),
 		);
-		const finalizedBlockSlot = this._chain.slots.getSlotNumber(finalizedBlock.timestamp);
-		const currentBlockSlot = this._chain.slots.getSlotNumber();
-		const threeRounds = this._chain.numberOfValidators * 3;
+		const validators = await this.blockExecutor.getValidators();
+		const finalizedBlockSlot = this.blockExecutor.getSlotNumber(finalizedBlock.timestamp);
+		const currentBlockSlot = this.blockExecutor.getSlotNumber(Math.floor(Date.now() / 1000));
+		const threeRounds = validators.length * 3;
```

Comment:
some of these values can be made constants?

Reply:
those values cannot be constant